x/globalfee/types/params.go

```diff
@@ -32,11 +35,65 @@ func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 	}
 }
 
+// this requires the fee non-negative
 func validateMinimumGasPrices(i interface{}) error {
 	v, ok := i.(sdk.DecCoins)
 	if !ok {
 		// todo check if err type correct
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "type: %T", i)
 	}
-	return v.Validate()
+
+	dec := DecCoins(v)
+	return dec.Validate()
+}
+
+// Validate checks that the DecCoins are sorted, have nonnegtive amount, with a valid and unique
+// denomination (i.e no duplicates). Otherwise, it returns an error.
+type DecCoins sdk.DecCoins
+
+func (coins DecCoins) Validate() error {
+	switch len(coins) {
+	case 0:
+		return nil
+
+	case 1:
+		// match the denom reg expr
+		if err := sdk.ValidateDenom(coins[0].Denom); err != nil {
+			return err
+		}
+		if coins[0].IsNegative() {
+			return fmt.Errorf("coin %s amount is negtive", coins[0])
```

Comment:
I think we want to accept 0 as an acceptable global min fee.
For instance at genesis when it is empty it will be equivalent to `0uatom`.
If we want to add a new consumer chain, we will want to allow the denom of that consumer chain to be accepted on the hub.
We may also still be OK with min fee being 0 of that new denom.
The param in that instance should be `0uatom,0ibc/asdfasdfasdf` so that `uatom` and the new `ibc/asdfasdfaf` denom can be used to pay fees.
Then it is up to each validator to increase the min fee in their app.toml, or a later proposal may decide to increase the min fee globally for one or both to something above 0.
we want to ensure it's still possible to have multiple denoms where one or more are 0.

Reply:
this logic seems to only be applied when there is one fee listed, which means the scenario I outlined above pass inside the `default` case below. That's great but we also want to ensure that a single 0 fee denom is still valid. For instance it might just be good housekeeping to make the `0uatom` global min fee explicit instead of relying on the default fallback value (maybe we don't want to use the `BondDenom` as the global min fee for instance).